### ISSUE TITLE
PR/commit hygiene concerns/suggestions

### ISSUE THREAD

**Issue by antstorm** [uber/cadence#3087]
Hey guys,

Really appreciate all the effort that is being put in to this project. It feels like Cadence is still super underrated and has enormous future potential.

My company is already using it for some use cases and we're planning on leveraging it much more. One important concern that our Security team has raised is around PR hygiene. It looks like lots of changes are being shipped without any description/context. I do understand that there's probably an internal process in place at Uber that coordinates these changes and there's probably plenty of discussions to support them, but I think exposing some of the context publicly would be very beneficial to Cadence users.

The use cases for this involve assessing whether upgrade is needed, figuring out how much potential risk it brings or finding a production issue related to a version upgrade.

Don't wanna impose too much bureaucracy here, but I think simple PR template would go a long way and will be greatly appreciated by other adopters of Cadence. It will also help open source devs to jump on board quicker and start contributing.

Basically the intention is to understand "What" has changed, "Why" it got changed and "How" it was tested/verified, with some risk scale assessment from the person making the change.

Hope that makes sense and please keep up the good work!
